packages/components/RadioGroup/src/RadioGroup.tsx

```diff
@@ -76,6 +76,26 @@ export const RadioGroup = compose<IRadioGroupType>({
       return null;
     }
 
+    // Generate ariaSetSize and ariaPosInSet for radio buttons if props are undefined
+    if (children) {
+      /* eslint-disable @typescript-eslint/ban-ts-ignore */
+      // @ts-ignore - TODO, fix typing error
+      children = React.Children.map(children, (child: React.ReactChild, index: number) => {
+        if (React.isValidElement(child)) {
+          const childProps = child.props;
+          const extraProps = {
+            ariaSetSize: childProps.ariaSetSize ?? React.Children.count(children),
+            ariaPosInSet: childProps.ariaPosInSet ?? index + 1,
+          };
+          return React.cloneElement(child, {
```

Comment:
Hmm... A little weary seeing "clone" and building a new prop object for every render. It seems like this could result in excessive rendering?

Looks like you could just pass in the "extraProps" part according to [their documentation](https://reactjs.org/docs/react-api.html#cloneelement): "The resulting element will have the original element’s props with the new props merged in shallowly." That might also help with render checks as posInSet and setSize are both numbers, so they should pass render checks, but the rest of the props are preserved.

Reply:
I think another strategy could be this:

Add `ariaSetSize: int` to RadioGroup's context, and set the value around line 81. Then, the children can just grab the value from RadioGroup's context and set their prop on the value.

I would also add something like: `ariaPosInSet: [ {radioButtonKey: string, ariaPosInSet: int} ]` to RadioGroup's context. Then, when iterating through the children using the map function, you can create a local array of these values. After .map is complete, you can set the context to the local array value. Using this, RadioButton's can grab this array, check for matching buttonKey, and set their posInSet to the corresponding value.

Reply:
That's a good point. I've been a little stumped on this because I don't know how to retrieve the children of RadioGroup outside of the render portion of this file. Ideally, I'd like to create a function that's defined in RadioGroup (maybe using [React.useEffect](https://reactjs.org/docs/hooks-reference.html#useeffect)) that will only be called when the number of children change and either passes down the data to the children as props or share the data through RadioGroupContext. But it appears we only know how many children there are in the render portion. And whatever I place there will have to run every render 😕

Reply:
I'll try to work with Sammy's suggestion and maybe add an array of the children in RadioGroupContext that will update with each render. The radio buttons can then pull the info they need from the array.

Reply:
I think if you remove the "childProps" part in cloneElement the children won't rerender excessively (at least, according to the documentation) but you'll still add the props you need to the children.

But yeah this is rough since the RadioGroup gets RadioButtons as children directly. I kind of wonder if/how the Picker handles this scenario since I think it has a similar structure.

Reply:
I ended up adding a `buttonKeys` array to RadioGroupContext and altered this if statement to retrieve the buttonKey for each child and place it into that array. I can then grab the info I need from the array from each child, eliminating the need to clone child elements and pass down props.